src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1444,12 +1447,14 @@ public AuditableRestAction<Void> kick(@Nonnull UserSnowflake user, String reason
             checkPosition(member);
 
         Route.CompiledRoute route = Route.Guilds.KICK_MEMBER.compile(getId(), user.getId());
+
+        AuditableRestAction<Void> action = new AuditableRestActionImpl<>(getJDA(), route);
         if (!Helpers.isBlank(reason))
         {
             Checks.check(reason.length() <= 512, "Reason cannot be longer than 512 characters.");
-            route = route.withQueryParams("reason", EncodingUtil.encodeUTF8(reason));
+            return action.reason(reason);
```

Comment:
I thought this didn't work for the kick and ban endpoints. was it updated to work?

Reply:
This is because of the [changelog](https://discord.com/developers/docs/change-log#api-v10):

> To specify a reason for an administrative action in audit logs, apps must now pass the X-Audit-Log-Reason header rather than the reason parameter for all endpoints. Read more in the [Audit Logs documentation](https://discord.com/developers/docs/resources/audit-log).

Reply:
Wow the v10 release post is surprisingly easy to miss in that changelog